File: src/main/java/com/back/together02be/global/util/TokenHashUtil.java
package com.back.together02be.global.util;

import java.nio.charset.StandardCharsets;
import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;
import java.util.Base64;

public final class TokenHashUtil {

    private TokenHashUtil() {
    }

    public static String sha256(String token) {
        try {
            MessageDigest digest = MessageDigest.getInstance("SHA-256");
            byte[] hashedToken = digest.digest(token.getBytes(StandardCharsets.UTF_8));
            return Base64.getUrlEncoder().withoutPadding().encodeToString(hashedToken);
        } catch (NoSuchAlgorithmException e) {
            throw new IllegalStateException("SHA-256 algorithm is not available.", e);
        }
    }
}

File: src/main/kotlin/com/back/together02be/global/util/TokenHashUtil.kt
package com.back.together02be.global.util

import java.nio.charset.StandardCharsets
import java.security.MessageDigest
import java.util.*

object TokenHashUtil {
    fun sha256(token: String): String {
        val digest = MessageDigest.getInstance("SHA-256")
        val hashedToken = digest.digest(token.toByteArray(StandardCharsets.UTF_8))

        return Base64.getUrlEncoder()
            .withoutPadding()
            .encodeToString(hashedToken)
    }
}
